packages/features/calendars/DatePicker.tsx

```diff
@@ -387,12 +406,14 @@ const DatePicker = ({
             datePickerDateActive: customClassNames?.datePickerDatesActive,
           }}
           weekStart={weekStart}
-          selected={selected}
+          selected={!isBookingInPast ? selected : null}
           {...passThroughProps}
           browsingDate={browsingDate}
           month={month}
           nextMonthButton={() => changeMonth(+1)}
-          slots={slots}
+          slots={!isBookingInPast ? slots : {}}
+          includedDates={!isBookingInPast ? includedDates : []}
+          isBookingInPast={isBookingInPast}
```

Comment:
Based on `isBookingInPast` we are modifying three other props that we are passing two same Component. I think it would make more sense to use pass all other props same and adjust those properties based on isBookingInPast inside the component.

In this way, to replicate this scenario of Booking In Past, one just needs to modify one prop and not 4 props.

Reply:
@anikdhabal Can we work on it in a followup. Merging now.